typescript-expressions/multiqc-tools/1.0.0/tests/multiqc-tools__1.0.0.test.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+// Author: Alexis Lucattini
+// For assistance on generation of typescript expression tests
+// In CWL, visit our wiki page at https://github.com/umccr/cwl-ica/wiki/TypeScript
+// Imports
+Object.defineProperty(exports, "__esModule", { value: true });
+var multiqc_tools__1_0_0_1 = require("../multiqc-tools__1.0.0");
+var MULTIQC_CL_CONFIG_STR_WITH_SAMPLENAME_REPLACE = JSON.stringify({
+    "sample_names_replace": {
+        "L1234567": "Ref_1_Good"
+    },
+    "another_cl_config_parameter": "foo"
+});
+var MULTIQC_CL_CONFIG_STR_WITHOUT_SAMPLENAME_REPLACE = JSON.stringify({
+    "another_cl_config_parameter": "foo"
+});
+var SAMPLE_NAME_OLD = "L13243546";
+var SAMPLE_NAME_NEW = "Ref_2_Bad";
+var EXPECTED_OUTPUT_MULTIQC_CL_CONFIG_STR_WITH_SAMPLENAME_REPLACE_OBJ = {
+    "sample_names_replace": {
+        "L1234567": "Ref_1_Good"
```

Comment:
Should this be "Ref_2_Bad" as in line18?

Reply:
No, the EXPECTED_OUTPUT_MULTIQC_CL_CONFIG_STR_WITH_SAMPLENAME_REPLACE_OBJ is 

```json
{
  "sample_names_replace": {
    "L1234567": "Ref_1_Good",
    "L13243546": "Ref_2_Bad"
  }
}
```

However if we were to place these variables in directly (which is what we'd do in typescript), we'd get an issue in ES5.1 where

```json
{
  "sample_names_replace": {
    "L1234567": "Ref_1_Good",
    SAMPLE_NAME_OLD: SAMPLE_NAME_NEW 
  }
}
```

would be evaluated to 

```json
{
  "sample_names_replace": {
    "L1234567": "Ref_1_Good",
    "SAMPLE_NAME_OLD": "SAMPLE_NAME_NEW"
  }
}
```

which is why we split the initialisation of the object into https://github.com/umccr/cwl-ica/pull/281/files/a0960f3f44277857ba3de094cc90dc28c8883f1c#diff-343a61231e67f24ca51d65ca1105f653c54d59b8fd4b8dc0e1eb424d2ae73002R19 AND then append the `"L13243546": "Ref_2_Bad"` here: https://github.com/umccr/cwl-ica/pull/281/files/a0960f3f44277857ba3de094cc90dc28c8883f1c#diff-343a61231e67f24ca51d65ca1105f653c54d59b8fd4b8dc0e1eb424d2ae73002R25

Reply:
Okay makes sense. Trying to understand what is the added value of this TS function as cl_config input to workflow is still the same and it's being passed on to this function along with output prefix.

Reply:
See screenshot above, as discussed on Wednesday, previously the sample name would have been "MDX210377_qualimap" but is now just "MDX210377".

Reply:
Noted and added a summary statement to this PR.